lib/metrics.js

```diff
@@ -15,67 +28,43 @@ module.exports = {
 function prepareData(res) {
   const audits = res.audits;
 
-  const resFMP = audits['first-meaningful-paint'];
-  const resFMPext = resFMP.extendedInfo;
-  const resSI = audits['speed-index-metric'];
-  const resSIext = resSI.extendedInfo;
-  const resTTI = audits['time-to-interactive'];
-  const resTTIext = resTTI.extendedInfo;
-
   const colorP0 = 'yellow';
   const colorP2 = 'green';
   const colorVisual = 'blue';
 
-  const timestamps = [{
-    title: 'Navigation Start',
-    name: 'navstart',
-    value: resFMPext && resFMPext.value.timestamps.navStart
-  }];
+  const timings = [];
+  metricsDefinitions.forEach(metric => {
+    try {
+      const resolvedMetric = {
+        title: metric.name,
+        name: metric.id,
+        value: metric.getTs(audits)
```

Comment:
so generally the flow is: get the timestamp for everything and then subtract off navStart. (and i think you'll also have to do a `/ 1000` to go from microseconds to milliseconds)

and it feels like this sorta changes the { timings, timestamps } return value at the end. 
a few ways to structure the data but i'd be okay with something like...

```js
{ 
  metrics: [
    { 
       name: 'Time to interactive',
       id: 'tti',
       timing: 2643.232,
       timestamp: 238932240584
    }, 
    // ... the other metrics...
   ],
   generatedTime: res.generatedTime,
   lighthouseVersion: res.lighthouseVersion,	
   initialUrl: res.initialUrl,
   url: res.url
}
```

but i'm open for other solutions.